src/ghostfat.c

```diff
@@ -94,19 +104,25 @@ static const struct TextFile info[] = {
 #define START_ROOTDIR (START_FAT1 + SECTORS_PER_FAT)
 #define START_CLUSTERS (START_ROOTDIR + ROOT_DIR_SECTORS)
 
+// all directory entries must fit in a single sector
+// because otherwise current code overflows buffer
+#define DIRENTRIES_PER_SECTOR (512/sizeof(DirEntry))
+STATIC_ASSERT(NUM_DIRENTRIES < DIRENTRIES_PER_SECTOR );
+
 static const FAT_BootBlock BootBlock = {
     .JumpInstruction = {0xeb, 0x3c, 0x90},
     .OEMInfo = "UF2 UF2 ",
     .SectorSize = 512,
     .SectorsPerCluster = 1,
     .ReservedSectors = RESERVED_SECTORS,
     .FATCopies = 2,
-    .RootDirectoryEntries = (ROOT_DIR_SECTORS * 512 / 32),
+    .RootDirectoryEntries = (ROOT_DIR_SECTORS * 512 / DIRENTRIES_PER_SECTOR),
```

Comment:
```suggestion
    .RootDirectoryEntries = (ROOT_DIR_SECTORS * DIRENTRIES_PER_SECTOR),
```

The computation was wrong causing trouble in samd21 bootloader